gui/middleware/zfs.py

```diff
@@ -59,6 +59,13 @@ def _vdev_type(name):
     return False
 
 
+def promote_zfs(name):
```

Comment:
I think you may have misunderstood me. When I said about middlewared I meant src/middlewared/middlewared/plugins/zfs.py

That should be part of `ZFSDatasetService`

Then we would use middleware.client to get the result. Let mw know if you need more directions on how to do it.